src/peerd/runtime.rs

```diff
@@ -296,7 +340,25 @@ impl Runtime {
                     &request.get_type()
                 );
                 self.messages_sent += 1;
-                self.sender.send_message(message)?;
+                while let Err(err) = self.peer_sender.send_message(message.clone()) {
+                    debug!("Error sending to remote peer in peerd runtime: {}", err);
+                    // If this is the listener-forked peerd, i.e. the maker's peerd, terminate it.
+                    if self.forked_from_listener {
+                        endpoints.send_to(
+                            ServiceBus::Ctl,
+                            self.identity(),
+                            ServiceId::Farcasterd,
+                            Request::PeerdTerminated,
+                        )?;
+                        warn!("Exiting peerd");
+                        std::process::exit(0);
+                    }
+
+                    if let Err(err) = self.reconnect_peer() {
```

Comment:
This works the way it is, because the outer while will catch any error with the peer runtime again. However, I think it would be more defensive if this and the other reconnect_peer call were a while loop as well. Changed in b41a7509fdb7cc0e60236f5b03614ee1bced029d .